sdk/identity/azure-identity/test/ut/client_certificate_credential_test.cpp

```diff
@@ -448,58 +448,6 @@ TempCertFile::TempCertFile()
   std::ofstream cert(Path, std::ios_base::out | std::ios_base::trunc);
 
   cert << // cspell:disable
-      "Bag Attributes\n"
```

Comment:
I don't remember why I removed that, will bring it back.

I remember though that the Windows APIs cannot read the certificates in this format. Trying to read them with .NET failed as well.

Reply:
FWIW, the windows crypto APIs basically don't handle PEM encoded data at all. It's quite annoying (and why I gave up making this change for the attestation APIs).

Reply:
Hmm, I think it will be a backwards compatibility problem, if the SDK was supporting `.pem` files and after updating to a newer version, they are no longer supported.
I appreciate the work that you've done @teo-tsirpanis, but I think while in this state, we won't be able to merge it. If you don't have more time to invest, the code in this PR will absolutely be useful for the next time that we attempt to remove the OpenSSL dependency. Otherwise, if you can somehow bring the `.pem` files support, let's continue. I may also take a shot at it, but later.

Reply:
Hmm, .NET has a separate method to import certificates from PEM which does the necessary conversions. I might write something in C++ but I don't know when I will have the time.

I will draft the PR again in the meantime.